src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -416,25 +118,26 @@ public DeploymentOptions()
             options.IgnoreKeywordCasing = false;
             options.IgnoreSemicolonBetweenStatements = false;
 
-            PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            // Initializing the default boolean type options to the BooleanOptionsDictionary
+            InitializeBooleanTypeOptions(options);
 
-            foreach (var deployOptionsProp in deploymentOptionsProperties)
+            // Initialize all non boolean properties
+            PropertyInfo[] deploymentOptionsProperties = this.GetType().GetProperties();
+            foreach (PropertyInfo deployOptionsProp in deploymentOptionsProperties)
             {
-                var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-
-                // Note that we set excluded object types here since dacfx has this value as null;
-                if (prop != null && deployOptionsProp.Name != "ExcludeObjectTypes")
-                {
-                    // Setting DacFx default values to the generic deployment options properties.
-                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
+                // Except ExcludeObjectTypes, as it has some STS defaults which needs to be considered here
```

Comment:
it would be helpful to add the line where ExcludeObjectTypes gets set in STS to this comment